### ISSUE TITLE
scoreline_i only carries two strings per line

### ISSUE THREAD

**Issue by joachimheintz** [csound/csound#443]
these lines are executed ok when written in the score section:
              i "bla" 0 1 "blu" "sound"
              i "bla" 1 1 "brown" "earth"
but when written in scoreline_i, only the first string is read.  scoreline behaves similar.

example below; uncomment the last score lines to see the difference.

```
<CsoundSynthesizer>
<CsOptions>
-nm0
</CsOptions>
<CsInstruments>
sr = 44100
nchnls = 1
ksmps = 32
0dbfs = 1

instr 1
scoreline_i {{
              i "bla" 0 1 "blu" "sound"
              i "bla" 1 1 "brown" "earth"
            }}
endin

instr bla
printf_i "p4 = %s, p5 = %s\n", 1, strget(p4), strget(p5)
endin

</CsInstruments>
<CsScore>
i 1 0 2
;              i "bla" 0 1 "blu" "sound"
;              i "bla" 1 1 "brown" "earth"
</CsScore>
</CsoundSynthesizer>
```

**Comment by kunstmusik** [comment]
I looked into this a little bit.  scoreline_i is receiving the correct string from the multi-line score string. That gets sent to csound->InputMessage() which also looks alright.  After that, sensline() looks to be processing correctly.  On strget, the values from:

```
    union {
      MYFLT d;
      int32 i;
    } ch;
```
are garbage values for p5.  I'm looking into this now to see if it has to do with the modified layout of pfields after RTTI was introduced.

**Comment by kunstmusik** [comment]
I was wrong about the garbage values, they were actually set correctly.  The problem was indeed in sensLine().  The location of where to index into the evtblk.strarg was being incorrectly calculated.  Because of the wrong calculation, it worked just fine for the first two string args that were set.  After that, any further string arguments were not being written into evtblk.strarg correctly.  The fix was to reset the offset of sstrp to the beginning of evtblk.strarg so that the code afterwards that fast-forwarded the pointer would work correctly. 

This fix is now in the develop branch and will be published in Csound 6.05.0.

**Comment by joachimheintz** [comment]
yes, works now.  thanks!
	joachim


Am 25.03.2015 um 20:37 schrieb Steven Yi:
